principal/options.go

```diff
@@ -384,3 +386,11 @@ func WithKeepAliveMinimumInterval(interval time.Duration) ServerOption {
 		return nil
 	}
 }
+
+func WithRedis(redisAddress, redisCompressionType string) ServerOption {
+	return func(o *Server) error {
+		o.options.redisAddress = redisAddress
+		o.options.redisCompressionType = redisCompressionType
```

Comment:
I feel we have a good chance to validate the compression type at this point.

Reply:
Yes, I am using an upstream function to parse compression type. Should I still validate the value while starting server?
https://github.com/jparsai/argocd-agent/blob/cluster/internal/argocd/cluster/cluster.go#L48

Reply:
Yes, I think the earlier the user receives feedback about erroneous configuration, the better.